### ISSUE TITLE
(basic & advanced) GUI items become stealable and lose functions

### ISSUE THREAD

**Issue by jaylawl** [Tuke-Nuke/TuSKe#14]
No errors are generated, here's what i did step by step:

Create this gui:

>                 create a gui with virtual chest with 2 rows named "Main Menu":
>                     set {_filler 1} and {_filler 2} to stained glass pane named "&r"
>                     set data value of {_filler 1} to 8
>                     set data value of {_filler 2} to 7
>                     loop 9 times:
>                         make next gui slot with {_filler 1}
>                     loop 9 times:
>                         make next gui slot with {_filler 2}
> 
>                     set {_solo} to unbreakable diamond hoe
>                     set data value of {_solo} to 101
>                     set 1st line of {_solo}'s lore to "&7Play a solo game"
>                     format gui slot 0 with {_solo} named "&f&lSolo":
>                         execute player command "/golf p solo"
> 
>                     format gui slot 17 with barrier named "&c&lClose":
>                         close player's inventory
> 
>                 open last gui to command sender

run server, test it, everything works as intended.

But then i edited some stuff in the same file and hit "/skript reload all", and afterwards the GUI lost its functionality (except for being displayed) and all items became stealable.

The same also happens with basic GUIs.

Restarting the server fixes this, but only until "skript reload all" is used once more.

Tuske 1.8.1
Skript dev30d

Update: usage of 1.8.2-dev4 did not fix this issue

**Comment by Blueyescat** [comment]
:white_check_mark: Confirmed with `latest 1.12.1 PaperSpigot`, `Skript 2.2-dev29` and `TuSKe 1.8.2 dev.4`.
`/sk reload scripts` (`/sk reload all` does the same) breaks it. Tried with more than one skript files or just 1 skript, still reloading all scripts breaks it. There is no problem with reloading aliases or config. But still you shouldn't use `/sk reload all/scripts` :slightly_smiling_face: `/sk reload <skript name>`

**Comment by jaylawl** [comment]
@Blueyescat but i tested it with both TuSKe latest stable (1.8.1) and the dev-4 ^^ is there any other version i haven't discovered yet?

"skript reload all" is super handy though ^^ can't really live without it, could you?

**Comment by WheezyGold7931** [comment]
@jaylawl just reload one skript at a time, no big deal....

**Comment by jaylawl** [comment]
@WheezyGold7931 it kinda is because having to type out all the script names can be very time consuming if your scripts are sorted into various sub-folders (which is inevitable with growing amounts)

just got this error while further messing around with the issue:

> [15:56:49 ERROR]: #!#!     at ch.njol.skript.SkriptEventHandler.check(Skrip
> [15:56:49 ERROR]: #!#!     at ch.njol.skript.SkriptEventHandler$1.execute(S
> [15:56:49 ERROR]: #!#!     at co.aikar.timings.TimedEventExecutor.execute(T
> [15:56:49 ERROR]: #!#!     at org.bukkit.plugin.RegisteredListener.callEven
> [15:56:49 ERROR]: #!#!     at org.bukkit.plugin.SimplePluginManager.fireEve
> [15:56:49 ERROR]: #!#!     at org.bukkit.plugin.SimplePluginManager.callEve
> [15:56:49 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_12_R1.event.CraftEv
> [15:56:49 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.EntityPlayer.cl
> [15:56:49 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_12_R1.entity.CraftH
> [15:56:49 ERROR]: #!#!     at ch.njol.skript.effects.EffOpenInventory.execu
> [15:56:49 ERROR]: #!#!     at ch.njol.skript.lang.Effect.run(Effect.java:52
> [15:56:49 ERROR]: #!#!     at ch.njol.skript.lang.TriggerItem.walk(TriggerI
> [15:56:49 ERROR]: #!#!     at ch.njol.skript.lang.TriggerItem.walk(TriggerI
> [15:56:49 ERROR]: #!#!     at com.github.tukenuke.tuske.util.EffectSection.
> [15:56:49 ERROR]: #!#!     at com.github.tukenuke.tuske.sections.gui.EffMak
> [15:56:49 ERROR]: #!#!     at com.github.tukenuke.tuske.manager.gui.v2.GUII
> [15:56:49 ERROR]: #!#!     at com.github.tukenuke.tuske.listeners.GUIListen
> [15:56:49 ERROR]: #!#!     at com.github.tukenuke.tuske.manager.gui.v2.Skri
> [15:56:49 ERROR]: #!#!     at ch.njol.skript.SkriptEventHandler.check(Skrip
> [15:56:49 ERROR]: #!#!     at ch.njol.skript.SkriptEventHandler$1.execute(S
> [15:56:49 ERROR]: #!#!     at co.aikar.timings.TimedEventExecutor.execute(T
> [15:56:49 ERROR]: #!#!     at org.bukkit.plugin.RegisteredListener.callEven
> [15:56:49 ERROR]: #!#!     at org.bukkit.plugin.SimplePluginManager.fireEve
> [15:56:49 ERROR]: #!#!     at org.bukkit.plugin.SimplePluginManager.callEve
> [15:56:49 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.PlayerConnectio
> [15:56:49 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.PacketPlayInWin
> [15:56:49 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.PacketPlayInWin
> [15:56:49 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.PlayerConnectio
> [15:56:49 ERROR]: #!#!     at java.util.concurrent.Executors$RunnableAdapte
> [15:56:49 ERROR]: #!#!     at java.util.concurrent.FutureTask.run(FutureTas
> [15:56:49 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.SystemUtils.a(S
> [15:56:49 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.MinecraftServer
> [15:56:49 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.DedicatedServer
> [15:56:49 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.MinecraftServer
> [15:56:49 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.MinecraftServer
> [15:56:49 ERROR]: #!#!     at java.lang.Thread.run(Thread.java:748)
> [15:56:49 ERROR]: #!#!
> [15:56:49 ERROR]: #!#! Version Information:
> [15:56:49 ERROR]: #!#!   Skript: 2.2-dev30d
> [15:56:49 ERROR]: #!#!   Bukkit: 1.12.1-R0.1-SNAPSHOT
> [15:56:49 ERROR]: #!#!   Minecraft: 1.12.1
> [15:56:49 ERROR]: #!#!   Java: 1.8.0_131 (Java HotSpot(TM) 64-Bit Server VM
> [15:56:49 ERROR]: #!#!   OS: Linux amd64 4.10.0-20-generic
> [15:56:49 ERROR]: #!#!
> [15:56:49 ERROR]: #!#! Running CraftBukkit: false
> [15:56:49 ERROR]: #!#! Running Spigot (or compatible): true
> [15:56:49 ERROR]: #!#! Running Paper (or compatible): true
> [15:56:49 ERROR]: #!#!
> [15:56:49 ERROR]: #!#! Current node: null
> [15:56:49 ERROR]: #!#! Current item: delete/clear last gui
> [15:56:49 ERROR]: #!#! Thread: Server thread
> [15:56:49 ERROR]: #!#!
> [15:56:49 ERROR]: #!#! Language: english
> [15:56:49 ERROR]: #!#! Link parse mode: DISABLED
> [15:56:49 ERROR]: #!#!
> [15:56:49 ERROR]: #!#! End of Error.
> [15:56:49 ERROR]: #!#!

afterwards the server was shut down automatically, cause unknown

**Comment by WhiteC4** [comment]
You could make a simple command that reloads all scripts one by one for now until there is a fix.

**Comment by jaylawl** [comment]
@WhiteC4 that's super ingenius, just did that. thanks for the heads up!

**Comment by Tuke-Nuke** [comment]
Since when you noticed that issue? After updating tuske, server, skript...?

**Comment by epicskymi** [comment]
I reported this via pm's but v25 of skript issue doesn't exist I updated to v29 of skript and server version 1.12 and 1.12.1 the issue exists and also affects skquery gui's

**Comment by epicskymi** [comment]
O wow this is intresting after ./skript reload all
it seems all your syntax of tuske have been unregistered 

because now the console just tells me cant understand all your gui manger syntaxs

**Comment by jaylawl** [comment]
i didnt update between minecraft versions 1.10 and now, 1.12.1 and their respective plugin versions - that's propably not helping the cause

**Comment by Tuke-Nuke** [comment]
Fixed it in latest dev build, please test it first to make sure it is really fixed (or any new issue).

**Comment by epicskymi** [comment]
what was the issue? b/c skquery will need to fix it aswell

**Comment by epicskymi** [comment]
it seems fixed nice

**Comment by Tuke-Nuke** [comment]
It looks like the issue is not related to new Skript updates, I think it should be there since gui manager remake. I don't know SkQuery's code but that's certainly a thing that won't be fixed.
The issue was that the gui listener was registered only once it was first used, but when you run `/sk reload all/scripts`, it disables that listener and doesn't register it again cause it thinks it still working.
I just used some Skript API methods to let me know when Skript will reload all scripts, so I can properly handle that.

**Comment by jaylawl** [comment]
Using dev6 this issue appears to be fully fixed, thanks for your time TukeNuke its appreciated! :)

**Comment by Tuke-Nuke** [comment]
Closing since it is fixed, thank's for reporting.
